### PR TITLE
MOVE_CORROSIVE_GAS Effect

### PR DESCRIPTION
**Pull request by AgustinGDLV**
Addresses issue #1099. Adds a battle script loosely modeled after Rototiller and Teeter Dance to implement Corrosive Gas's effect. Haze is used as a placeholder animation. Behavior now mirrors that of Knock Off in Gen 5+. Note that Wild Pokémon cannot knock off player items; right now, I've translated this to mean you cannot corrode your own items in a double wild battle. Any corroded non-berry items are given back after battle.

Corrosive Gas

<details>
  <summary>Old Issues/TODO:</summary>

**Issues:** (resolved)
- [x] Magic Coat / Magic Bounce currently has no effect against Corrosive Gas in a double battle.
- [x] Magic Coat / Magic Bounce does not reflect Corrosive Gas unless the original use had a valid target.
- [x] When Magic Bounce activates, the original user's ability pops up instead of the target's Magic Bounce.
- [x] If Corrosive Gas is used by a Pokémon with Prankster, it shouldn't affect Dark types.
- [x] Find more edge cases, apparently.

**In-Game Behavior:**
- [ ]  Test if a new item tricked to a Pokémon affected by Corrosive Gas retains its effects (or if an item can even be tricked) in Sword and Shield. Currently, this PR effectively places Embargo on every affected Pokémon until the end of battle.
- [ ]  Test if Knock Off has a damage bonus against Pokémon with corroded items.
- [ ]  Test behavior with Magic Coat / Magic Bounce in a double battle in Sword and Shield.

</details>

## **Discord Contact Info**
Agustin#1522

### CODE REVIEW

**Comment by ghoulslash** [comment]
Don't forgot to add a corroded item check to `AI_GetHoldEffect` as well

**Comment by AgustinGDLV** [comment]
No more glaring issues. New commit doubles down on Bulbapedia's statement that Corrosive Gas has the same effect as Knock Off in Gen 3 and 4. I have removed the `corrodedItems` bitfield and now use the same bitfield that Knock Off does. This means that Knock Off does not have a damage boost on Pokémon with corroded items and Pokémon with corroded items cannot give or receive items from Trick or Bestow. With regards to the latter, this function was not originally implemented for old Knock Off, but I think this is because Bestow is a Gen 5 move and I am making the assumption that this is how it is supposed to work for both moves.

The behavior with Magic Bounce in double battles matches other multi-target status moves (I tested Growl). It might still be necessary to double check both this and the above functionality in-game, as Smogon claims that Corrosive Gas removes the target's item entirely like Knock Off in Gen 5+. If it does, it probably shouldn't be terribly difficult to copy from #2130.

File: src/battle_script_commands.c [hunk 1]
```diff
@@ -9580,6 +9581,55 @@ static void Cmd_various(void)
...
+            CheckSetUnburden(gActiveBattler);
+            gBattleStruct->itemStolen[gBattlerPartyIndexes[gActiveBattler]].stolen = TRUE;
+            BtlController_EmitSetMonData(BUFFER_A, REQUEST_HELDITEM_BATTLE, 0, sizeof(gBattleMons[gActiveBattler].item), &gBattleMons[gActiveBattler].item);
+            MarkBattlerForControllerExec(gActiveBattler);
+            gBattlescriptCurrInstr += 7;
+        }
+        return;
+    }
+    case VARIOUS_JUMP_IF_CANT_USE_CORROSIVE_GAS:
+        for (gBattlerTarget = 0; gBattlerTarget < gBattlersCount; gBattlerTarget++)
```
Comment (ghoulslash) [inline src/battle_script_commands.c:9609]: it is safer to use `i` or similar variable instead of looping over gBattlerTarget directly.

**Comment by DizzyEggg** [comment]
Replaced by #3152
